scripts/pull-gee-repository.sh

```diff
@@ -0,0 +1,370 @@
+#===============================================================================
+function ssh_or_https ()
+{
+    printf "Do you want to clone via (h)ttps or (s)sh (default is https)? " 
+    read clonetype
+    if [ "$clonetype" = "s" ]
+    then
+        return $IS_SSH
+    fi   
+    return $IS_HTTPS
+}
+
+#===  FUNCTION  ================================================================
+#          NAME:  get_password
+#   DESCRIPTION:  prompts user to enter password and sets global
+#    PARAMETERS:  n/a
+#       RETURNS:  n/a
+#===============================================================================
+function get_password ()
+{
+    printf "What is your password? "
+    read -s PASSWORD 
+}
+
+#===  FUNCTION  ================================================================
+#          NAME:  get_username
+#   DESCRIPTION:  prompts user to enter username and sets global
+#    PARAMETERS:  n/a
+#       RETURNS:  n/a
+#===============================================================================
+function get_username ()
+{
+    printf "What is your username? "
+    read USER_NAME 
+}
+
+#===  FUNCTION  ================================================================
+#          NAME:  get_dir
+#   DESCRIPTION:  allows the user to set the directory name to clone into
+#    PARAMETERS:  n/a
+#       RETURNS:  n/a
+#===============================================================================
+
+function get_dir ()
+{
+    printf "Which directory do you wish to clone into (default is ./earthenterprise) ? "
+    read DIR
+    if [ -n "$DIR" ]
+    then
+        DIR_NAME=$DIR
+    fi
+}
+
+#===  FUNCTION  ================================================================
+#          NAME:  print_help
+#   DESCRIPTION:  prints help info
+#    PARAMETERS:  n/a
+#       RETURNS:  n/a
+#===============================================================================
+function print_help ()
+{
+    cat <<MSG
+
+${THIS} [-d <dir>] [-f <fork>] [-h] [-i] [-p <password>]
+
+    
+    -d|--dir <dir>
+        Directory under which to clone into (default is "earthenterprise")
+    
+    -f|--fork <fork>
+        Clone from forked repository. Default is SSH
+
+    -h|--help
+        Show this help message and exit.
+
+    -i|--interactive
+        Allows for user interaction
+
+    -p|--password <password>
+        User password. Only forked repositories cloned in HTTPS use this
+
+MSG
+}
+
+#===  FUNCTION  ================================================================
+#          NAME:  interactive_mode
+#   DESCRIPTION:  allows for user interaction
+#    PARAMETERS:  n/a
+#       RETURNS:  n/a
+#===============================================================================
+
+function interactive_mode ()
+{
+    get_dir
+    get_type
+    CLONE_REPO=$?
+
+    if is_cloned
+    then
+        echo "$DIR_NAME is already cloned!"
+        exit 1 
+    fi
+    
+    # if cloning main, just use standard https
+    if [ $CLONE_REPO -eq $IS_MAIN ]
```

Comment:
I'm curious why "cloning main" ==> "want to use https"?  Personally, I only ever want to use ssh, to save on the number of passwords I need to keep up with.

Also, why does the path a little lower go through get_password, but this one doesn't?

Reply:
This was talked about it slack, current build instructions list as ssh but it was figured that https would be best if you are only cloning the main (google/earthenterprise) repo.  Just covering the case where the user does not have a github account and just wants to pull down the repo.  Thus, no password is required.

Cloning a fork is a different case

Reply:
Ah, interesting - I hadn't recalled that "anonymous https" was an option ;).